ddtrace/internal/telemetry/writer.py

```diff
@@ -0,0 +1,250 @@
+import time
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ...internal import forksafe
+from ...settings import _config as config
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..runtime import get_runtime_id
+from ..service import ServiceStatus
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .data import get_application
+from .data import get_host_info
+
+
+log = get_logger(__name__)
+
+
+def _get_interval_or_default():
+    return float(get_env("instrumentation_telemetry", "interval_seconds", default=60))
+
+
+class TelemetryWriter(PeriodicService):
+    """
+    Periodic service which sends Telemetry request payloads to the agent.
+    Supports version one of the instrumentation telemetry api
+    """
+
+    # telemetry endpoint uses events platform v2 api
+    ENDPOINT = "telemetry/proxy/api/v2/apmtelemetry"
+
+    def __init__(self, agent_url=None):
+        # type: (Optional[str]) -> None
+        super(TelemetryWriter, self).__init__(interval=_get_interval_or_default())
+
+        self.enabled = False  # type: bool
```

Comment:
```suggestion
        self._enabled = False  # type: bool
```

should probably make `enabled` a property so it can be guarded with the lock if it does need to be exposed publicly

Reply:
`enabled` is only acced inside of a lock. Guarding it with the telemetry lock will result in a deadlock. I can try to refactor the enable()/disable() logic to make this work

Reply:
It doesn't look liked a public `enabled` is required so we're good here 👍